server/services/visitSessionsService.ts

```diff
@@ -65,55 +61,50 @@ export default class VisitSessionsService {
-                : visitSession.closedVisitCapacity - visitSession.closedVisitBookedCount,
-            capacity: visitRestriction === 'OPEN' ? visitSession.openVisitCapacity : visitSession.closedVisitCapacity,
-            visitRoomName: visitSession.visitRoomName,
-            sessionConflicts: visitSession.sessionConflicts,
-            visitRestriction,
+        // Month grouping for slots, e.g. 'February 2022'
+        const slotMonth = format(parsedStartTimestamp, 'MMMM yyyy')
+        if (!slotList[slotMonth]) slotList[slotMonth] = [] // eslint-disable-line no-param-reassign
+
+        // Slots for the day, e.g. 'Monday 14 February'
+        const slotDate = format(parsedStartTimestamp, 'EEEE d MMMM')
+        let slotsForDay: VisitSlotsForDay = slotList[slotMonth].find(slot => slot.date === slotDate)
+        if (slotsForDay === undefined) {
+          slotsForDay = {
+            date: slotDate,
+            slots: { morning: [], afternoon: [] },
+            prisonerEvents: { morning: [], afternoon: [] },
           }
+          slotList[slotMonth].push(slotsForDay)
+        }
 
-          // Maybe this needs doing below, twice since technically this may not be pushed
-          if (isAfter(parsedEndTimestamp, latestEndTime)) {
-            latestEndTime = parsedEndTimestamp
-          }
-          if (isBefore(parsedStartTimestamp, earliestStartTime)) {
-            earliestStartTime = parsedStartTimestamp
-          }
+        const newSlot: VisitSlot = {
+          id: (slotIdCounter + 1).toString(),
+          startTimestamp: visitSession.startTimestamp,
+          endTimestamp: visitSession.endTimestamp,
+          availableTables:
+            visitRestriction === 'OPEN'
+              ? visitSession.openVisitCapacity - visitSession.openVisitBookedCount
+              : visitSession.closedVisitCapacity - visitSession.closedVisitBookedCount,
+          capacity: visitRestriction === 'OPEN' ? visitSession.openVisitCapacity : visitSession.closedVisitCapacity,
+          visitRoomName: visitSession.visitRoomName,
+          sessionConflicts: visitSession.sessionConflicts,
+          visitRestriction,
+        }
 
-          // Add new Slot to morning / afternoon grouping
-          if (parsedStartTimestamp.getHours() < this.morningCutoff) {
-            if (timeOfDay === '' || timeOfDay === 'morning') {
-              slotsForDay.slots.morning.push(newSlot)
-            }
-          } else if (timeOfDay === '' || timeOfDay === 'afternoon') {
-            slotsForDay.slots.afternoon.push(newSlot)
-          }
+        // Maybe this needs doing below, twice since technically this may not be pushed
+        if (isAfter(parsedEndTimestamp, latestEndTime)) {
+          latestEndTime = parsedEndTimestamp
+        }
+        if (isBefore(parsedStartTimestamp, earliestStartTime)) {
+          earliestStartTime = parsedStartTimestamp
+        }
+
+        // Add new Slot to morning / afternoon grouping
```

Comment:
`timeOfDay` doesn't exist so is always the equivalent of empty so no need for the `if` statements.